src/Logic.hs

```diff
@@ -722,12 +727,12 @@ parseMergeCommand projectConfig triggerConfig = cvtParseResult . P.parse pCommen
     -- configured.
     pDeployToEnvironment :: Parser DeployEnvironment
     pDeployToEnvironment
-      | (defaultEnvironment : _) <- environments
+      | length environments == 0 = fail noDeployEnvironmentsError
       -- Without the try this could consume the space and break 'merge and deploy on friday'
-      = P.try (P.hspace1 *> pString "to" *> P.hspace1) *> P.choice pDeployEnvironments
-      <|> pure (DeployEnvironment defaultEnvironment)
-      | otherwise
-      = fail noDeployEnvironmentsError
+      | length environments == 1 = P.try (P.hspace1 *> pString "to" *> P.hspace1 *> P.choice pDeployEnvironments)
+        <|> pure (DeployEnvironment $ head environments)
+      | otherwise = P.hspace1 *> pString "to" *> P.hspace1 *> P.choice pDeployEnvironments
+        <|> fail noDeployEnvironmentSpecifiedError
```

Comment:
Performance in Hof isn't super important and especially not since when there are only ever a couple branches configured, but could you change this to the following?

Haskell's lists are lazy linked lists, so `length xs` will traverse the entire `xs` list before it can tell you the length. In this case pattern matching is both more performant (you only ever have to inspect the first two elements for the pattern match) and more idiomatic (you can also get rid of `head` this way).

```haskell
    pDeployToEnvironment :: Parser DeployEnvironment
    pDeployToEnvironment =
      -- Without the try this could consume the space and break 'merge and deploy on friday'
      if null environments
        then fail noDeployEnvironmentsError
        else P.try (P.hspace1 *> pString "to" *> P.hspace1 *> P.choice pDeployEnvironments)
               <|> defaultEnvironment

    -- The default environment to deploy to on a "merge and deploy". This
    -- behavior is deprecated when more than one environment is set, and will
    -- return an error in that case.
    defaultEnvironment :: Parser DeployEnvironment
    defaultEnvironment
      | [environment] <- environments = pure (DeployEnvironment environment)
      | otherwise = fail noDeployEnvironmentSpecifiedError
```

Reply:
This reads a lot nicer indeed, thanks :)